frame/system/src/lib.rs

```diff
@@ -1913,6 +1913,44 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn signed_ext_check_weight_refund_works() {
+		new_test_ext().execute_with(|| {
+			let info = DispatchInfo { weight: 512, ..Default::default() };
+			let post_info = PostDispatchInfo { actual_weight: Some(128), };
+			let len = 0_usize;
+
+			AllExtrinsicsWeight::put(256);
+
+			let pre = CheckWeight::<Test>(PhantomData).pre_dispatch(&1, CALL, &info, len).unwrap();
+			assert_eq!(AllExtrinsicsWeight::get().unwrap(), info.weight + 256);
+
+			assert!(CheckWeight::<Test>::post_dispatch(
+				pre, &info, &post_info, len, &Ok(())).is_ok()
+			);
```

Comment:
needs reformat.

Reply:
(this one is seriously not really readable :D)

Reply:
at least parentheses opened in the same line should be closed in the same line IMO.